lib/bambora/bank/batch_report_resource.rb

```diff
@@ -53,17 +53,24 @@ def initialize(client:, api_key:)
       #
       #  @params profile_data [Hash] with values as noted in the example.
       def show(report_data)
-        add_messages_to_response(
-          client.post(path: sub_path, body: batch_report_body(report_data)),
-        )
+        response = client.post(path: sub_path, body: batch_report_body(report_data))
+
+        response = ensure_record_key_exists(response)
+        response = add_messages_to_response(response)
+
+        response
       end
 
       private
 
-      def add_messages_to_response(response)
-        # bambora can return null or empty record results, don't decorate with messages
-        return response if response.dig(:response, :record).nil?
+      def ensure_record_key_exists(response)
+        # bambora can return null or empty record results, fill it in for consistency
+        response.dig(:response)[:record] = [] if response.dig(:response, :record).nil?
```

Comment:
Instead of returning the response with no `:record`, decorate the response with it.